.github/workflows/ci.yaml

```diff
@@ -0,0 +1,32 @@
+name: CI build
+on:
+  pull_request:
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.14
+      - uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - name: Ensure gofmt
+        run: test -z "$(gofmt -s -d cmd pkg)"
+      - name: Ensure go.mod is already tidied
+        run: go mod tidy && git diff --no-patch --exit-code
+      - name: Run unit tests
+        run: go test ./...
```

Comment:
I like the idea of tidying the go modules and checking for gofmt, but can you put those in the `Makefile`? We already have [Prow doing the tests](https://github.com/kubernetes/test-infra/blob/f34ec251f8b943950bbd0718e13c47b661ea97da/config/jobs/kubernetes-sigs/aws-encryption-provider/presubmits.yaml), and I want to keep the CI similar to the dev workflow (just `make test`).

I think running goreleaser with `--snapshot --skip-publish --rm-dist` on PRs in GH workflow is reasonable